### PR TITLE
Add broker timeouts

### PR DESCRIPTION
**Pull request by konstantin-msft**

### CODE REVIEW

File: change/@azure-msal-common-558c780e-3387-4832-8c07-b5449309b5dc.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "Add broker timeouts #5580",
+  "packageName": "@azure/msal-common",
+  "email": "sh.konst89@gmail.com",
```
Comment (tnorling) [inline change/@azure-msal-common-558c780e-3387-4832-8c07-b5449309b5dc.json]: Please update to your Microsoft email

File: lib/msal-common/src/telemetry/performance/PerformanceEvent.ts [hunk 2]
```diff
@@ -221,6 +221,12 @@ export type StaticFields = {
     matsHttpEventCount?: number;
     httpVerToken?: string;
     httpVerAuthority?: string;
+
+    // Broker timeouts
```
Comment (tnorling) [inline lib/msal-common/src/telemetry/performance/PerformanceEvent.ts]: Can you add a bit more context here? i.e. "This is used by the 1P browser lib". I do wonder though if there's a way we can avoid exposing these types of details to the 3P/common libs at all?

File: change/@azure-msal-common-558c780e-3387-4832-8c07-b5449309b5dc.json (hunk 1 above)
Comment (konstantin-msft) [reply change/@azure-msal-common-558c780e-3387-4832-8c07-b5449309b5dc.json]: Good catch, thank you! Fixed

File: lib/msal-common/src/telemetry/performance/PerformanceEvent.ts (hunk 2 above)
Comment (konstantin-msft) [reply lib/msal-common/src/telemetry/performance/PerformanceEvent.ts]: Not at the moment unfortunately, but we should be able in the future by extending 3p telemetry classes in 1p. Added context.

File: lib/msal-common/src/telemetry/performance/PerformanceEvent.ts [hunk 3]
```diff
@@ -221,6 +221,12 @@ export type StaticFields = {
     matsHttpEventCount?: number;
     httpVerToken?: string;
     httpVerAuthority?: string;
+
+    // Broker timeouts used by 1p browser lib
+    brokerInteractionTimeout?: number;
```
Comment (peterzenz) [inline lib/msal-common/src/telemetry/performance/PerformanceEvent.ts]: I'm assuming these are all in ms?  Suggest renaming them to brokerInteractionTimeoutMs, etc.

**Review by peterzenz — APPROVED** [review]

File: lib/msal-common/src/telemetry/performance/PerformanceEvent.ts (hunk 3 above)
Comment (konstantin-msft) [reply lib/msal-common/src/telemetry/performance/PerformanceEvent.ts]: Good point, thank you. Done.

**Review by jo-arroyo — APPROVED** [review]

**Review by tnorling — APPROVED** [review]

**Pull request merged**